### PR TITLE
bug(settings): Handle incorrect email case in sign up

### PR DESCRIPTION
**Pull request by dschom**
## Because

- We did not handle the incorrect email properly, and would just display incorrect email to the user.
- The incorrect email error contains a corrected email in the payload that should be used.

## This pull request

- Retries the login with the email provided by the incorrect email error state.
- Fixes typescript error in functional-tests/pages/config.ts.
- Adds functional test to cover this edge case.

## Issue that this pull request solves

Closes: FXA-9663

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

This also impacted the sign in unblock flow and is now fixed. However, it might conflict with https://github.com/mozilla/fxa/pull/16959  :disappointed: .

### CODE REVIEW

File: packages/functional-tests/pages/config.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export class ConfigPage extends BaseLayout {
       const metaConfig = page.locator('meta[name="fxa-content-server/config"]');
       const config = await metaConfig.getAttribute('content');
 
-      this.config = JSON.parse(decodeURIComponent(config));
+      this.config = JSON.parse(decodeURIComponent(config || '{}'));
```
Comment (dschom) [inline packages/functional-tests/pages/config.ts:22]: fixes typescript error / warning

File: packages/fxa-graphql-api/src/gql/lib/error.ts [hunk 2]
```diff
@@ -43,6 +43,8 @@ export function CatchGatewayError(
             code: err.code,
             errno: err.errno,
             info: err.info,
+            // Important, used for 'Incorrect email case' error
+            email: err.email,
```
Comment (vbudhram) [inline packages/fxa-graphql-api/src/gql/lib/error.ts:47]: 👍🏽

**Review by vbudhram — APPROVED** [review]
@dschom LGTM, can't wait to move to V2 creds where this won't be an issue
